### PR TITLE
Blaze: Remove the stubbed response for campaign creation request and enabled the feature flag

### LINKED ISSUE #11803: Blaze: Analytics
- [x] #11804
- [x] #11812
- [x] #11816
- [x] #11835
- [x] #11841
- [x] Once API is ready test steps from #11840

### LINKED ISSUE #11894: Remove stubbed response when the APIs are ready
- [x] AI suggestions
- [x] Campaign creation
- [x] Payment info

### PR DESCRIPTION
**Pull request by pmusolino**
Closes: #11894
Closes: #11803

## Description
This PR updates the campaign creation endpoint handling since the endpoint has been deployed, removing the stubbed response. Additionally, I have activated the feature flag to make the feature accessible in the production environment.

## Testing instructions
1. Test that you can create a new Blaze campaign in production.
2. Based on [this issue](https://github.com/woocommerce/woocommerce-ios/issues/11803), follow the testing instructions of [this PR](https://github.com/woocommerce/woocommerce-ios/pull/11840) for Campaign Creation, checking that the tracking is working as expected.
3. Make sure you are able to create a new Blaze campaign starting from Menu -> Blaze.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/191">17.9 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR contains changes to <code>RELEASE-NOTES.txt</code>.<br />
Note that these changes won't affect the final version of the release notes as this version is in code freeze.<br />
Please, get in touch with a release manager if you want to update the final release notes.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by JorgeMucientes** [comment]
Hey @pmusolino , I tested the feature and noticed 2 issues. One is critical, and the other I think it can be fixed for the next release.

- **Critical**: when creating a campaign, the API request will succeed, but when attempting to parse the response, the App fails to parse the JSON. The app will show and error in the screen, but the campaign will be created successfully. Checking the app logs I can see the following: 
```
Swift.DecodingError.typeMismatch(Swift.String, Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: \"id\", intValue: nil)], debugDescription: \"Expected to decode String but found number instead.\", underlyingError: nil)
```
The API is returning an `int` instead of a `string`. According to the contract it should be a `string`. I reached out to the team to see if they can fix it: p1711467136152709-slack-C069RG07W6S In Android we also expect a `string` but it doesn't fail because our parsing library is not that strict and it parses the integer correctly to string. Hopefully no changes are required here. We just need to keep track of the API change before merging this.

- The second problem is related to the start date. We had the same problem in Android and I fixed it today here: https://github.com/woocommerce/woocommerce-android/pull/11163. The start date must be set to the current day + 1 day. For example if today is 26 of Mach we have the ensure the start date is at least set to 27 of March. If the current day is used to create the campaign, the campaign will be created successfully and immediately canceled.

**Comment by pmusolino** [comment]
Thanks, @JorgeMucientes for testing it.

Let me know when the API will be fixed.

In the meantime: 

> The second problem is related to the start date. We had the same problem in Android and I fixed it today here: https://github.com/woocommerce/woocommerce-android/pull/11163. The start date must be set to the current day + 1 day. For example if today is 26 of Mach we have the ensure the start date is at least set to 27 of March. If the current day is used to create the campaign, the campaign will be created successfully and immediately canceled.

I fixed this issue: a544f4ba425c22c78f646dc8c1910bc5bf605487

I also found another bug, not allowing users to create a new campaign starting from Menu -> Blaze. This was also fixed here: d7c11cb269c9b5d2511b50a58edfd264749098fd

**Comment by JorgeMucientes** [comment]
Hey @pmusolino I did another round of testing and everything is working well now. The API is also fixed now and campaigns can be created successfully. 
I'll let any of the other iOS devs add their review before merging. Thanks for tackling the fixes.

File: WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift
Comment (JorgeMucientes) [inline WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift:32]: Just curious about this. Is it a common pattern in iOS to work with seconds for time, instead of millis?

**Review by JorgeMucientes — APPROVED** [review]

File: WooCommerce/Classes/ViewRelated/Blaze/BudgetSetting/BlazeBudgetSettingView.swift
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/BudgetSetting/BlazeBudgetSettingView.swift]: Maybe we can make this as a public static constant and use it from `BlazeCampaignCreationFormViewModel` for defining the initial `startDate` value.

**Review by selanthiraiyan — APPROVED** [review]
Thanks for working on this, Paolo. 

I was able to create a Blaze campaign from the app following the native flow. 

I left a non-blocking nit. :shipit:

File: WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift
Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift:32]: I would say yes. The `Date` structure represents a specific point in time. When performing date and time calculations, such as adding or subtracting time intervals, it's common to use seconds as the unit of time. This is because the `Date` structure's underlying representation is based on a time interval in seconds from a reference date.

In this specific case, the Foundation framework provides a type `TimeInterval`, which is a typealias for `Double` and represents a time interval in seconds. This makes it convenient to perform time calculations in seconds. For example, to add one day to the current date as I did, you would add 86400 seconds (60 seconds * 60 minutes * 24 hours) to the current `Date`.

In any case, you can also work with milliseconds if you want, but not common in iOS.

File: WooCommerce/Classes/ViewRelated/Blaze/BudgetSetting/BlazeBudgetSettingView.swift
Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Blaze/BudgetSetting/BlazeBudgetSettingView.swift]: I applied the changes in this commit d6d4866474df3113298f8928db2806680390ab9b

I moved it under `BlazeBudgetSettingViewModel`, since it's related just to the Budget view. Plus, this variable is not definiting the start date, but the minimum day allowed in the picker, since the start date could be also different from the minimum day allowed (eg. editing a campaign). 👍

**Pull request merged**
